content/ui-patterns/messaging.mdx

```diff
@@ -5,27 +5,159 @@ description: Messaging components are used to provide important and relevant inf
 
-- _Flash alerts are best suited for this type of messaging._
+(image here)
+
+## Message states
+
+<img
+  width="960"
+  alt="A group of labels showing the proper icon and color combination for each message state"
+  src="https://github.com/primer/css/assets/18661030/41d3a706-9aa3-4f05-8bad-5176f49a7d5d"
+/>
+
+Messaging components combine state with message type to provide the appropriate level of prominence. To determine which component to use for your message, start by identifying the message state. From there, follow the decision tree to pick a component best suited for your use case.
+
+### Info
+
+Use the `info` state to highlight messages that help users complete a task or that provide additional context. This state is only used for information that is not critical to the user's experience.
+
+Examples:
+- a setting that impacts the current view
+- a nudge towards a new feature
+- a process is in progress
+- single sign on to access data
+
+<img
+  width="960"
+  alt="draft wip"
+  src="https://github.com/primer/css/assets/18661030/3fa52b6d-7e33-4f27-b0ec-d836375fe21a"
+/>
+
+(image of banner in context)
+(image of callout in context)
+
+### Warning
+
+Use the `warning` state to inform users of a potential issue, an important upcoming change that may impact their experience or a potentially unexpected consequence of an action.
+
+Examples:
+- an account setting will become required soon
+- a degraded experience(link) due to connectivity or GitHub availability
+- a non-destructive action with side-effects like costs or access changes is about to take place
+
+<img
+  width="960"
+  alt="draft wip"
+  src="https://github.com/primer/css/assets/18661030/ae82693a-e18c-425f-9f63-e2077012e76b"
+/>
+
+(image of banner in context)
+(image of callout in context)
+(image of announcement in context)
+
+### Success
+
+Use the `success` state to notify users that the results of an action were successful if it is not apparent from the UI.
+
+Examples:
+- saving an account setting if its otherwise not clear the save has taken place
+- an issue has been transferred successfully
+
+<img
+  width="960"
+  alt="draft wip"
+  src="https://github.com/primer/css/assets/18661030/a3612e09-97b2-4ff5-903a-e85d1587ce50"
+/>
+
+(image of banner in context)
+(image of inline message in context)
+
+### Unavailable
+
+Reserve the `unavailable` state for **inline** system related degraded experiences.
+
+Examples:
+- a row of a data table could not be loaded
+- an item in a list could not be loaded
+
+See [degraded experiences](./degraded-experiences) for more information.
+
+### Critical
+
+Use the `critical` state to notify users that the results of an action were unsuccessful, or to inform that an action is destructive. This state is typically used for form errors and confirmation dialogs.
+
+Examples:
+- form field is invalid upon submitting
```

Comment:
This is a validation error, no? This should be presented as an inline validation error, no?

Reply:
Yes, that is what we are talking about here. Inline validation error == inline message of type `critical`